draft-ietf-quic-transport.md

```diff
@@ -3874,145 +3874,134 @@ The details of loss detection and congestion control are described in
-support or does not have access to received ECN codepoints, it does not increase
-ECN counts.
-
-Coalesced packets (see {{packet-coalesce}}) mean that several packets can share
-the same IP header.  The ECN counts for the ECN codepoint received in the
-associated IP header are incremented once for each QUIC packet, not per
-enclosing IP packet or UDP datagram.
+detect and respond to network congestion.  ECN allows an endpoint to set an ECT
+codepoint in the ECN field of an IP packet. A network node can then indicate
+congestion by setting the CE codepoint in the ECN field instead of dropping the
+packet {{?RFC8087}}.  Endpoints react to reported congestion by reducing their
+sending rate in response, as described in {{QUIC-RECOVERY}}.
 
-Each packet number space maintains separate acknowledgement state and separate
-ECN counts.  For example, if one each of an Initial, 0-RTT, Handshake, and 1-RTT
-QUIC packet are coalesced, the corresponding counts for the Initial and
-Handshake packet number space will be incremented by one and the counts for the
-application data packet number space will be increased by two.
+To enable ECN, a sending QUIC endpoint first determines whether a path supports
+ECN marking and whether the peer reports the ECN values in received IP headers;
+see {{ecn-validation}}.
 
 
-### ECN Validation {#ecn-validation}
+### Reporting ECN Counts
+
+Use of ECN requires the receiving endpoint to read the ECN field from an IP
+packet, which is not possible on all platforms. If an endpoint does not
+implement ECN support or does not have access to received ECN field, it
+does not report ECN counts for packets it receives.
+
+Even if an endpoint does not set an ECT field on packets it sends, the endpoint
+MUST provide feedback about ECN markings it receives, if these are accessible.
+Failing to report the ECN counts will cause the sender to disable use of ECN
+for packets to this receiver.
+
+On receiving an IP packet with an ECT(0), ECT(1) or CE codepoint, an
+ECN-enabled endpoint accesses the ECN field and increases the corresponding
+ECT(0), ECT(1), or CE count. These ECN counts are included in subsequent ACK
+frames; see {{generating-acks}} and {{frame-ack}}.
 
-It is possible for faulty network devices to corrupt or erroneously drop packets
-with ECN markings.  To provide robust connectivity in the presence of such
-devices, each endpoint independently validates ECN counts and disables ECN if
-the path is not showing consistent support for ECN.
+Each packet number space maintains separate acknowledgement state and separate
+ECN counts.  Coalesced QUIC packets (see {{packet-coalesce}}) share the same IP
+header so the ECN counts are incremented once for each QUIC packet.
+
+For example, if one each of an Initial, Handshake, and 1-RTT QUIC packet are
+coalesced into a single UDP datagram, the ECN counts for all three packet
+number spaces will be incremented by one each.
```

Comment:
```suggestion
number spaces will be incremented by one each.  Due to when keys are discarded, it's extremely unlikely a packet with all 4 types will arrive in the same datagram, but this forms a complete example.
```

Reply:
Isn't this based on old text?  This example lists three types only, which is possible (our stack can do three at the server in some cases).